packages/react-vapor/src/components/validation/hoc/WithNonEmptyValueMultiSelectValidationHOC.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {createStructuredSelector} from 'reselect';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IMultiSelectOwnProps} from '../../select/MultiSelectConnected';
+import {SelectSelector} from '../../select/SelectSelector';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationTypes} from '../ValidationTypes';
+import {InferableComponentEnhancer} from './connectHOC';
+
+const mapStateToProps = createStructuredSelector({
+    selectedValues: SelectSelector.getMultiSelectSelectedValues,
+});
+
+const mapDispatchToProps = (dispatch: IDispatch) => ({
+    setError: (id: string, error: string) => dispatch(ValidationActions.setError(id, error, ValidationTypes.nonEmpty)),
+    clearError: (id: string) => dispatch(ValidationActions.clearError(id, ValidationTypes.nonEmpty)),
+});
+
+export interface IWithNonEmptyValueMultiSelectValidationProps {
+    validationMessage?: string;
```

Comment:
I think I would call it `nonEmptyErrorMessage` instead, so that it doesn't clash with other HOCs

Reply:
The other validations i saw used this naming convention `IWithNonEmptyValueInputValidationProps` and `IMultiSelectWithDirtyOwnProps`. not sure if i should but the `NonEmpty` or the `MultiSelect` first?

Reply:
I would remove the I before, we have few of them here and there that shouldn't be there, but our guidelines tells not to https://github.com/coveord/tsjs#interface

Reply:
😕 if we use both `WithNonEmpty` and `WithDirty` (and we will), the `validationMessage` will need to be different for both cases, so we really need a different prop.

This is kind of a flaw in the current HOCs...

Maybe we should always include the purpose of the message from now on?